.ruff.toml

```diff
@@ -1,38 +1,69 @@
 # Enable the pycodestyle (`E`) and Pyflakes (`F`) rules by default.
 # Unlike Flake8, Ruff doesn't enable pycodestyle warnings (`W`) or
 # McCabe complexity (`C901`) by default.
-select = ["E", "F", "W", "I", "UP", "ASYNC", "YTT", "A", "COM", "C4", "T10", "EM", 
-"FA", "ISC", "PIE", "PYI", "Q", "RSE", "RET", "SLOT", "SIM", "TID", "TCH", "INT", 
-"PD", "PGH", "PLC", "PLE", "PLW", "FLY", "NPY", "PERF", "B", "DTZ", "ANN"]
+lint.select = [
+    "A",
+    "ANN",
+    "ASYNC",
+    "B", # bugbear 
+    "C4",
+    "COM",
+    "DTZ",
+    "E", # pycodestyle
+    "EM",
+    "F", # Pyflakes
+    "FA",
+    "FLY",
+    "I",
+    "INT",
+    "ISC",
+    "NPY",
+    "PD",
+    "PERF",
+    "PGH",
+    "PIE",
+    "PLC",
+    "PLE",
+    "PLW",
+    "PYI",
+    "Q",
+    "RET",
+    "RSE",
+    "S", # Bandit
+    "SIM",
+    "SLOT",
+    "T10",
+    "TCH",
+    "TID",
+    "UP",
+    "W", # pycodestyle warnings
+    "YTT",
+]
 
+# Enabled to allow direct type comparisons using is and is not under E721 (see https://docs.astral.sh/ruff/rules/type-comparison/)
+lint.preview = true
 
 # ignore E501 - linelength limit (covered by black except in docstrings) 
 # and PD901 - use of df variable name
-ignore = ["E501", "PD901", "ANN101"]
+lint.ignore = ["E501", "PD901", "ANN101"]
```

Comment:
Now we have removed black we should make sure to raise an issue to get line length checking back in soon since it is being ignored in ruff (E501).  However i think not having this is worth it for the other improvements/streamlining of the CI this change brings.  Potentially this could be addressed in #169

Reply:
Added ticket #190. And yes agreed that we might look at it as part of #169